src/OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.cpp

```diff
@@ -85,6 +85,68 @@ const State StateBuilder::buildState(const Instant& anInstant, const VectorXd& a
     return {anInstant, aCoordinates, this->frameSPtr_, this->coordinatesBrokerSPtr_};
 }
 
+const State StateBuilder::build(
+    const State& aState, const Map<const Shared<const CoordinatesSubset>, const VectorXd>& anAdditionalCoordinatesMap
+) const
+{
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("StateBuilder");
+    }
+
+    if (!aState.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("State");
+    }
+
+    if (aState.accessFrame() != this->frameSPtr_)
+    {
+        throw ostk::core::error::runtime::Wrong("Frame");
+    }
+
+    VectorXd coordinates = VectorXd(this->coordinatesBrokerSPtr_->getNumberOfCoordinates());
+    Integer nextIndex = 0;
+
+    for (const auto& subset : this->coordinatesBrokerSPtr_->getSubsets())
+    {
+        Integer subsetDetections = 0;
+        VectorXd subsetCoordinates;
+
+        if (aState.accessCoordinatesBroker()->hasSubset(subset))
+        {
+            subsetDetections++;
+            subsetCoordinates = aState.extractCoordinates(subset);
+        }
+
+        const auto coordinatesMapIterator = anAdditionalCoordinatesMap.find(subset);
+        if (coordinatesMapIterator != anAdditionalCoordinatesMap.end())
+        {
+            subsetDetections++;
+            subsetCoordinates = coordinatesMapIterator->second;
+        }
+
+        if (subsetDetections == 0)
+        {
+            throw ostk::core::error::RuntimeError("Missing CoordinatesSubset");
```

Comment:
```suggestion
            throw ostk::core::error::RuntimeError(String::Format("Missing CoordinatesSubset [{}]", *subset));
```

Reply:
Added these. P.S., the RuntimeError class has built-in string formatting capabilities, so you can do just:
`RuntimeError("Missing CoordinatesSubset [{}]", subset->getName())`